packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -92,7 +92,7 @@ function BlockWrapper( {
 			marginHorizontal,
 		},
 	];
-	const accessible = ! ( isSelected || isInnerBlockSelected );
+	const accessible = ! ( isSelected || isDescendentBlockSelected );
```

Comment:
I'm concerned `accessible` relying upon a deep check might cause [performance issues](https://github.com/WordPress/gutenberg/pull/44992#issuecomment-1279478405). However, from testing just now, I do not observe any specific issue. Will you explore Voice Over yourself to see if issues arise for you? 

If there are no issues, including this change would fix https://github.com/WordPress/gutenberg/issues/45415 and supersede https://github.com/WordPress/gutenberg/pull/44992#issuecomment-1279478405. 🎉

Reply:
Oh I didn't know those issues existed 😅 When I was testing I realized I couldn't move between inner blocks for both iOS and Android that's why I added the `deep` check. I tested it when I made the change but let me double-check again following the other issue instructions as well and report back. 👍

Reply:
I can confirm I don't see any performance issues when using Voice Control or Voice over on an iPhone 14 using the demo app in dev mode.